src/widgets/latest-blocks/LatestBlocks.vue

```diff
@@ -47,34 +47,14 @@ import BaseLink from '@/shared/ui/components/BaseLink.vue';
 import BaseButton from '@/shared/ui/components/BaseButton.vue';
 import BaseContentBlock from '@/shared/ui/components/BaseContentBlock.vue';
 import BaseLoading from '@/shared/ui/components/BaseLoading.vue';
-import type { Block } from '@/shared/api/schemas';
-import { onMounted, ref, shallowRef } from 'vue';
-import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
+import { computed } from 'vue';
 import TimeStamp from '@/shared/ui/components/TimeStamp.vue';
+import { setupAsyncData } from '@/shared/utils/setup-async-data';
 
-const emit = defineEmits<{
-  loaded: [number]
-}>();
+const setup = setupAsyncData(() => http.fetchBlocks({ per_page: 10 }));
```

Comment:
Again, look at how composables are defined in VueUse. You usually can use them in a way like this:

```ts
const { isLoading, data: blocks } = setupAsyncData(...) 
``` 

Consider aligning your composable with what's an established standard in the ecosystem.